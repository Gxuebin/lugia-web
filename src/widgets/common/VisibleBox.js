/*
 * 可通过visible属性控制是否可见的盒子
 * @flow
 */
import * as React from 'react';
import styled from 'styled-components';

type VisibleBoxProps = {
  visible: boolean
}
export default styled.div`
   ${(props: VisibleBoxProps) => (props.visible ? '' : 'display: none;')}
`;
