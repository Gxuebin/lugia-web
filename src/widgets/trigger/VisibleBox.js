/*
 * 可通过visible属性控制是否可见的盒子
 */
import React from 'react';
import styled from 'styled-components';

export default styled.div`
  display: ${props => (props.visible ? '' : 'none')};
`;
