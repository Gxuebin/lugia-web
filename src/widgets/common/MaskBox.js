/*
 * 遮罩盒子
 * @flow
 */
import * as React from 'react';
import VisibleBox from '../common/VisibleBox';
import styled from 'styled-components';

export const ZIndex = 1050;
export default styled(VisibleBox)`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55, 55, 55, 0.6);
  height: 100%;
  filter: alpha(opacity=50);
  z-index: ${ZIndex};
`;
