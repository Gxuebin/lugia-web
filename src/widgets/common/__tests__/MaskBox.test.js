//@flow
import React from 'react';
import chai from 'chai';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import VisibleBox from '../VisibleBox';
import MaskBox from '../MaskBox';

const { expect: exp } = chai;

describe('VisibleBox', () => {

  it('visible  is true', () => {
    const maskBox = renderer.create(<MaskBox visible></MaskBox>).toJSON();
    expect(maskBox).toMatchSnapshot();
  });


});
