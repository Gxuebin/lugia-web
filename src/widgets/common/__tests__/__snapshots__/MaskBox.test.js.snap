// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibleBox visible  is true 1`] = `
.c0 {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: #373737;
  background-color: rgba(55,55,55,0.6);
  height: 100%;
  -webkit-filter: alpha(opacity=50);
  filter: alpha(opacity=50);
  z-index: 1050;
}

<div
  className="c0"
/>
`;
